sdlbook.c

```diff
@@ -729,27 +752,37 @@ int main(int argc, char **argv) {
 							goto dun_goofed;
 						case SDLK_KP_PLUS:
 							need_redraw = change_scale(+10);
+							change_scroll_h(+96);
+							change_scroll_v(+64);
 							break;
 						case SDLK_KP_MINUS:
 							need_redraw = change_scale(-10);
+							change_scroll_h(-96);
+							change_scroll_v(-64);
 							break;
 						case SDLK_PAGEDOWN:
-							need_redraw = change_scroll(+page_dims.h);
+							need_redraw = change_scroll_v(+page_dims.h);
 							break;
 						case SDLK_PAGEUP:
-							need_redraw = change_scroll(-page_dims.h);
+							need_redraw = change_scroll_v(-page_dims.h);
 							break;
 						case SDLK_UP:
 							if((event.mod & KMOD_LCTRL) || (event.mod & KMOD_RCTRL))
-								need_redraw = change_scroll(-96);
+								need_redraw = change_scroll_v(-96);
 							else
-								need_redraw = change_scroll(-32);
+								need_redraw = change_scroll_v(-32);
 							break;
 						case SDLK_DOWN:
 							if((event.mod & KMOD_LCTRL) || (event.mod & KMOD_RCTRL))
-								need_redraw = change_scroll(+96);
+								need_redraw = change_scroll_v(+96);
 							else
-								need_redraw = change_scroll(+32);
+								need_redraw = change_scroll_v(+32);
+							break;
+						case SDLK_LEFT:
```

Comment:
shouldn't these 2 also use the "CTRL held" logic for bigger jumps ?

Reply:
agree via 7d49319ded28796e516b723e385f043a217472ca !